Processing_data/Program.cs

```diff
@@ -11,13 +10,20 @@
     Console.WriteLine("Failed to deserialize the JSON data");
     return;
 }
-PrintingStatsService printingService = new PrintingStatsService();
-StatsBucket statsBucket = new StatsBucket(printingService, 5);
 
-foreach (DataValue dataValue in data)
+var query = from d in data
+            group d by new Coordinate() { Latitude = Math.Round(d.Latitude, 2), Longitude = Math.Round(d.Longitude, 2) } into g
+            select new { g.Key, Value = new Statistics(g.Select(x => x.Value).ToList()) };
+
+// var query = data.GroupBy(x => new Coordinate() { Latitude = Math.Round(x.Latitude, 2), Longitude = Math.Round(x.Longitude, 2) })
+//                 .Select(g => new { Key = g.Key, Value = new Statistics(g.Select(x => x.Value).ToList()) });
+
+
+PrintingStatsService printingService = new PrintingStatsService();
+StatsBucket statsBucket = new StatsBucket();
+foreach (var item in query)
```

Comment:
try to use Aggregate method from Linq. (there is one of the limitations of the expression syntax).